relay-server/src/services/processor.rs

```diff
@@ -2555,11 +2556,16 @@ impl EnvelopeProcessorService {
                         // always result in a request, otherwise we would enter an endless loop.
                         self.send_global_partition(partition_key, &mut partition);
                         remaining = Some(next);
+                        partition_overflowed = true;
                     }
                 }
             }
         }
 
+        if partition_overflowed {
+            metric!(counter(RelayCounters::PartitionOverflow) += 1);
+        }
```

Comment:
I think a distribution telling us into how many parts one flush was split would be interesting. This would also not be skewed by how many times we flushed.

Reply:
Yeah I could add this instead